sdk/openai/openai-rest/samples-dev/streamChatCompletions.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to list chat completions for a chat context.
+ *
+ * @summary list chat completions.
+ * @azsdk-weight 100
+ */
+
+import createClient, { AzureKeyCredential, ChatCompletionsOutput } from "@azure-rest/openai";
+import { createSseStream } from "@azure/core-sse";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+import { IncomingMessage } from "http";
+dotenv.config();
+
+// You will need to set these environment variables or edit the following values
+const endpoint = process.env["ENDPOINT"] || "<endpoint>";
+const azureApiKey = process.env["AZURE_API_KEY"] || "<api key>";
+
+export async function main() {
+  console.log("== Streaming Chat Completions Sample ==");
+
+  const client = createClient(endpoint, new AzureKeyCredential(azureApiKey));
+  const deploymentName = "gpt-35-turbo";
+  const response = await client
+    .path("/deployments/{deploymentId}/chat/completions", deploymentName)
+    .post({
+      body: {
+        messages: [
+          { role: "system", content: "You are a helpful assistant. You will talk like a pirate." },
+          { role: "user", content: "Can you help me?" },
+          { role: "assistant", content: "Arrrr! Of course, me hearty! What can I do for ye?" },
+          { role: "user", content: "What's the best way to train a parrot?" },
+        ],
+        stream: true,
+      },
+    })
+    .asNodeStream();
+
+  const stream = response.body;
+  if (!stream) {
+    throw new Error("The response stream is undefined");
+  }
+
+  if (response.status !== "200") {
+    throw new Error(`Failed to get chat completions: ${streamToString(stream)}`);
+  }
+
+  const sses = createSseStream(stream as IncomingMessage);
```

Comment:
This cast doesn't feel ideal but I am open to ideas on how to improve the situation. We could add back an overload for `NodeJS.ReadableStream` in core-sse and dynamically check if [`socket`](https://nodejs.org/api/http.html#responsesocket) exists but I wouldn't prefer that.